src/splitter.rs

```diff
@@ -5,11 +5,90 @@ impl Splitter {
         Splitter
     }
 
-    pub fn vocabulary(&self, _text: &str) -> Vec<String> {
-        todo!("Implement vocabulary extraction")
+    pub fn vocabulary(&self, text: &str) -> Vec<String> {
+        use std::collections::BTreeSet;
+        
+        text.split_whitespace()
+            .map(|word| word.to_lowercase())
+            .collect::<BTreeSet<_>>()
+            .into_iter()
+            .collect()
     }
 
     pub fn phrases(&self, _text: &str) -> Vec<Vec<u16>> {
         todo!("Implement phrases extraction")
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_vocabulary_no_duplicates() {
+        let splitter = Splitter::new();
+        let text = "hello world hello rust world";
+        let vocab = splitter.vocabulary(text);
+        
+        // Check that there are no duplicates by comparing length with unique count
+        let unique_count = vocab.iter().collect::<std::collections::HashSet<_>>().len();
```

Comment:
This is a bit clever for a unit test. Just hard code the expected length in the test. Don't do calculations in tests beyond that required for the "act" part.